iceoryx2-tunnel/conformance-tests/Cargo.toml

```diff
@@ -0,0 +1,27 @@
+[package]
+name = "iceoryx2-tunnel-conformance-tests"
+description = "iceoryx2: Conformance tests for iceoryx2-tunnel"
+categories = { workspace = true }
+edition = { workspace = true }
+homepage = { workspace = true }
+keywords = { workspace = true }
+license = { workspace = true }
+repository = { workspace = true }
+rust-version = { workspace = true }
+version = { workspace = true }
+
+[features]
+default = ["tunnel_zenoh"] # TODO this should use whatever 'iceoryx2-tunnel/tunnel_zenoh' has set
+tunnel_zenoh = ["iceoryx2-tunnel/tunnel_zenoh"] # TODO same here
+
+[dependencies]
+iceoryx2-bb-conformance-test-macros = { workspace = true }
+iceoryx2 = { workspace = true }
+iceoryx2-bb-posix = { workspace = true }
+iceoryx2-bb-testing = { workspace = true }
+iceoryx2-tunnel = { workspace = true, optional = true }
+iceoryx2-tunnel-backend = { workspace = true }
+iceoryx2-services-discovery = { workspace = true }
+
+[dev-dependencies]
+iceoryx2-tunnel-zenoh = { workspace = true, optional = false } # TODO: this should be optional and but dev-dependencies cannot be optional
```

Comment:
@orecham since you are currently gaining quite some experience with the feature propagation, do you have a proposal how to cleanly fix the TODOs?